api/src/main/java/io/opencensus/tags/TagMetadata.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2019, OpenCensus Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opencensus.tags;
+
+import com.google.auto.value.AutoValue;
+import javax.annotation.concurrent.Immutable;
+
+/**
+ * {@link TagMetadata} contains properties associated with a {@link Tag}.
+ *
+ * <p>For now only the property {@link TagTtl} is defined. In future, additional properties may be
+ * added to address specific situations.
+ *
+ * @since 0.20
+ */
+@Immutable
+@AutoValue
+public abstract class TagMetadata {
+
+  TagMetadata() {}
+
+  /**
+   * {@link Tag}s associated with this metadata can propagate unlimited number of hops.
+   *
+   * @since 0.20
+   */
+  public static final TagMetadata METADATA_UNLIMITED_PROPAGATION =
```

Comment:
Not sure we want these 2 fields public, because if we add other things in the metadata will be hard to scale to all possibilities. But I see why you want to expose this to avoid object creation, but user may be able to do this as well, not sure what is the best option.